src/template/template.ts

```diff
@@ -51,7 +52,11 @@ export default abstract class Template {
             const doc = await fs.readFile(templatePath, 'utf-8');
             const namespace = await this.getNamespace(filePath);
 
-            let text = doc
+            let text = doc;
+
+            text = await fileScopedNamespaceConverter.getFileScopedNamespaceFormOfTemplateIfNecessary(text, filePath);
```

Comment:
This should be abstracted away from this piece of code (makes it difficult to read). But that is an issue in the repo as a whole, and therefore not an issue for this PR